client/src/ui-kit/toast.js

```diff
@@ -0,0 +1,81 @@
+import styled, { css } from "styled-components";
+import { variables } from "./variables";
+import React, {Component} from "react";
+
+const StyledToast = styled.div`
+	position:fixed;
+	top:2em;
+	left:50%;
+	transform:translate(-50%,0);
+	max-width:600px;
+	min-width: 320px;
+  border-radius: ${variables.button.borderRadius};
+  padding: 20px 15px 20px 70px;
+  line-height:1.5;
+  background: ${variables.colour.paleGrey};
+  color: ${variables.colour.offBlack};
+  box-shadow: 4px 4px 12px ${variables.colour.lightGrey};
+
+  &:before {
+    color:white;
+    height:50px;
+    width:50px;
+    position:absolute;
+    left:8px;
+    top: 50%; 
+    transform: translate(0,-50%);
+    font-size:30px;
+    padding-top:2px;
+    border-radius:${variables.circle};
+  }  
+
+  ${props => props.type==="success" && css`
+	  &:before {
+	  	background:#6cd577;
+	  	background:linear-gradient(to bottom right, #6cd577,#b6eec9);
+	    content: "✔";
+	  }    
+  `}
+
+  ${props => props.type==="error" && css`
+	  &:before {
+	  	background:#eb6773;
+	  	background:linear-gradient(to bottom right, #eb6773,#f3b2ba);
+	    content: "✘";
+	  }   
+  `}
+`;
+
+class Toast extends Component {
+
+	constructor(props) {
+    super(props);
+    this.state = {look: true};
+  }
+
+	disappear = (e) => {
+		this.setState({look:false})
+	};
+
+	componentDidMount() {
```

Comment:
I just tested out the toast, I think instead of calling the function in componentDidMount, it should be called in componentWillReceiveProps because i was passing in controlled state to the database feedback component but it was not re-rendering. If i understand correctly, we dont want the toast to show on initial render but only when users have submitted their rating and success props are sent.

Reply:
Totally! Thank you for this feedback. Once I pull our latest version tonight at school, let's talk how we want it, and I'll make it again. Also, I'm liking Della's solution with Redux instead of local state:

Alright, so a few things!

The first thing I want to say is that your final solution is fine for what you want to do! I have an aversion to using stateful components, but sometimes even I buckle and use them. This case is completely reasonable for doing so.

That being said, there are options here!

First, you say this:

>I don’t believe that disappearing should be sent to Redux state. It should disappear on its own, like, the disappearing is more part of its style than function.

This is a challenging problem that happens a lot when dealing with externalized state - when do we want components to have their own internal state? If you feel very strongly about this component having it's own timeout state, than I appreciate that you wouldn't want this to live in the redux state, but in CASE you wanted to, let me show you how it would be done.

First, for components like this, I want to emphasis there are like half a dozen different styles and methods to write it out - here I'll show you one that requires the least amount of change while also relying on redux state. I imagine right now you have DBFeedback rendering somewhere, like

```<div className="header"> 
   <DbFeedback />
</div>
...```

Or something like that, lets walk through the actions that would have to transpire to have it visible, and how to tie that into redux.

1. Some sort of event needs to fire. In this case, I'll make a few assumptions about how your `review` action looks, something like this:

```const userReview = (locationID, locationName) => dispatch =>
fetch('backendUrl', { //I don't know if you're using fetch, it doesn't matter anything is fine here
    method: 'post',
    body: JSON.stringify({locationID, locationName})
}).then(function(response) {
    return response.json();
  })
.then(success => {
   dispatch({type: 'SHOW_TOAST', payload: {type: 'error', message: `Successfully saved location ${locationName}` } }); //or some action like this
   setTimeout(() => dispatch({type: 'HIDE_TOAST'  } }), 3000);
})
.catch(err => {
  dispatch({type: 'SHOW_TOAST', payload: { type: 'error', message: `Successfully saved location ${locationName}` } }); 
  setTimeout(() => dispatch({type: 'HIDE_TOAST' } }), 5000); //lets pretend you want errors to take longer
})```


then in your reducer, something like this:


```// I'll pretend this is in a reducer called toast
case 'SHOW_TOAST':
   return {...state, toastIsVisible: true, toastMessage: action.payload.message, toastType: action.payload.type};

case 'HIDE_TAST':
   return {...state, toastIsVisible: false, toastMessage: null, toastType: null} // nulling out these other values is not really necessary, probably```


And then in component that is displaying the toast:


```const MyHeader = props =>
<div className="header">
  {props.showToast && <DbFeedback message={props.toastMessage} type={props.toastType} />}
</div>

// lets assume this component is hooked up to state
export connect((state) => ({ toastMessage: state.toast.toastMessage, showToast: state.toast.toastIsVisible, toastType: state.toast.toastType }))(MyHeader)```

Reply:
I already have the API call to the backend in my code, we are using axios!

Reply:
That's just what Della was writing without having seen our code.